framework/wazuh/core/tests/test_utils.py

```diff
@@ -1998,3 +2000,68 @@ def test_agents_allow_higher_versions(new_conf, agents_conf):
         else:
             with pytest.raises(exception.WazuhError, match=".* 1129 .*"):
                 utils.check_agents_allow_higher_versions(new_conf)
+
+
+@patch('wazuh.core.utils.tempfile.mkstemp', return_value=('handle', os.path.join(OSSEC_TMP_PATH, 'file.tmp')))
+@patch('wazuh.core.utils.chmod')
+@patch('wazuh.core.common.wazuh_gid')
+@patch('wazuh.core.common.wazuh_uid')
+def test_upload_file(mock_uid, mock_gid, mock_chmod, mock_mkstemp):
+    """Test upload_file function."""
+
+    mko = mock_open()
+    filename = "file.xml"
+    with patch('wazuh.core.utils.open', mko):
+        # Test upload simple content.
+        content = "Test"
+        with patch('wazuh.core.utils.safe_move') as mock_safe_move:
+            result = utils.upload_file(content, file_path=filename, check_xml_formula_values=False)
+            handle = mko()
+            assert isinstance(result, WazuhResult)
+            handle.write.assert_called_once_with(content)
+
+            tmp_path = os.path.join(OSSEC_TMP_PATH, 'file.tmp')
+            file_path = os.path.join(WAZUH_PATH, filename)
+            mock_safe_move.assert_called_once_with(tmp_path,
+                                                          file_path,
+                                                          ownership=(mock_uid(), mock_gid()),
+                                                          permissions=0o660)
+
+    mko = mock_open()
+    with patch('wazuh.core.utils.open', mko):
+        # Test upload formula content check_xml=True.
+        content = "<Test>+Tes't</Test>"
+        with patch('wazuh.core.utils.safe_move') as mock_safe_move:
+            result = utils.upload_file(content, file_path=filename, check_xml_formula_values=True)
+            handle = mko()
+            assert isinstance(result, WazuhResult)
+            handle.write.assert_called_once_with("<Test>'+Tes't</Test>")
+
+            tmp_path = os.path.join(OSSEC_TMP_PATH, 'file.tmp')
+            file_path = os.path.join(WAZUH_PATH, filename)
+            mock_safe_move.assert_called_once_with(tmp_path,
+                                                          file_path,
+                                                          ownership=(mock_uid(), mock_gid()),
+                                                          permissions=0o660)
```

Comment:
Very few things change between the validations, to not repeat everything twice I would do something like

```py
@pytest.mark.parametrize('content, check_xml, called_with', [
    ('Test', False, 'Test'),
    ('<Test>+Tes't</Test>', True, '<Test>'+Tes't</Test>')
])
```

And use the parameters to call the same functions

Reply:
The test_upload_file function was parameterized for each test case.